wazo_websocketd/bin/init_amqp.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+# Copyright 2019 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import argparse
+import sys
+
+import kombu
+
+ROUTING_KEYS = [
+    'applications.#',
+    'auth.#',
+    'call_log.#',
+    'calls.#',
+    'chatd.#',
+    'collectd.#',
+    'conferences.#',
+    'config.#',
+    'directory.#',
+    'faxes.#',
+    'plugin.#',
+    'service.#',
+    'status.#',
+    'switchboards.#',
+    'sysconfd.#',
+    'voicemails.#',
+]
+
+
+def _parse_cli_args(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--username',
+        action='store',
+        default='guest',
+        help="The username to use to connect to RabbitMQ",
+    )
+    parser.add_argument(
+        '--password',
+        action='store',
+        default='guest',
+        help="The password to connect to RabbitMQ",
+    )
+    parser.add_argument(
+        '--port', action='store', default='5672', help="The port of RabbitMQ"
+    )
+    parser.add_argument(
+        '--host', action='store', default='localhost', help="The host of RabbitMQ"
+    )
+    parser.add_argument(
+        '--upstream_exchange_name',
+        action='store',
+        default='xivo',
+        help="The upstream exchange name",
+    )
+    parser.add_argument(
+        '--upstream_exchange_type',
+        action='store',
+        default='topic',
+        help="The upstream exchange type",
+    )
+    parser.add_argument(
+        '--exchange_name',
+        action='store',
+        default='wazo-websocketd',
+        help="The main exchange name",
+    )
+    parser.add_argument(
+        '--exchange_type',
+        action='store',
+        default='headers',
+        help="The main exchange type",
+    )
+    return parser.parse_args(args)
+
+
+def main():
+    config = _parse_cli_args(sys.argv[1:])
+    bus_url = 'amqp://{username}:{password}@{host}:{port}//'.format(
+        username=config.username,
+        password=config.password,
+        host=config.host,
+        port=config.port,
+    )
+    upstream_exchange = kombu.Exchange(
+        config.upstream_exchange_name,
+        type=config.upstream_exchange_type,
+        auto_delete=False,
+        durable=True,
+        delivery_mode='persistent',
+    )
+    exchange = kombu.Exchange(
+        config.exchange_name,
+        type=config.exchange_type,
+        auto_delete=False,
+        durable=True,
+        delivery_mode='persistent',
+    )
+
+    with kombu.Connection(bus_url) as connection:
```

Comment:
Since this is always run by systemd, maybe we should move it websocketd itself.

Reply:
No objection, it was copied from a script which was extracted into script to be extracted in another service
But in this case, the queue/bindings are 100% owned by websocketd
In all case, we should move forward with exchange header and get rid of this :grin:

Reply:
> Since this is always run by systemd, maybe we should move it websocketd itself.

I moved the creation of the exchange to the startup of the websocketd process. This implementation is synchronous before starting the event loop. Does that sound right to you or should I do it later during the startup process or lazily when the first Session adds its bindings?

Reply:
At startup that looks good to me.
I was thinking about future container and want to avoid having a run.sh script that does ton of preparation in a run.sh entrypoint script.